packages/signature-controller/src/SignatureController.ts

```diff
@@ -499,33 +544,13 @@ export class SignatureController extends BaseControllerV2<
         messageManager.setMessageStatusSigned(messageId, signature);
       }
 
-      this.#acceptApproval(messageId);
-
-      return this.#getAllState();
+      return signature;
     } catch (error: any) {
       console.info(`MetaMaskController - ${methodName} failed.`, error);
-      this.#errorMessage(messageManager, messageId, error.message);
```

Comment:
We need this call and the `#errorMessage` method to ensure that the status is updated if signing fails, and that the typed messages use the error status while the others mark the messages as rejected.

Reply:
This also suggested we need additional unit tests to verify the status is correctly updated if signing fails on each message type.

Reply:
Ok, I will add this back.
This also reminded me why I didn't remove `RejectRequest` from this controller. Are we cool to reject approval from this controller when this case happens? Because sign operations happen via `newUnsignedXMessage` methods, and these functions are not exposed in the UI.

Reply:
We don't need to reject the approval anymore as it had to have already been approved to get to this point. We just need to ensure any errors during signing update the status of the message accordingly.

Reply:
Alright, that's true, approval will be resolved regardless of the result of this signing operation.
Thanks Matthew